### PR TITLE
Implement iterating/collecting PauliSumOp coefficients #5547

### LINKED ISSUE #5547: implement iterating/collecting PauliSumOp coefficients
It would be useful to be able to iterate over, or return a list of coefficients in PauliSumOp. At present, one has to dig into the details to construct the coefficients for each term. In particular, it requires multiplying two numbers for each coeffcient. There are use cases, for instance when bounding the eigenvalues of the operators.

For example, this gives a bound on the eigenvalues:
```python
float(sum(abs(pauli_sum.primitive.coeffs)) * abs(pauli_sum.coeff))
```

Abstracting access would be cleaner and resistant to implementation changes. Like this, if we want to return an numpy array
```python
float(sum(abs(pauli_sum.coeffs)))
```
or this, if we return a list
```python
sum(abs(c) for c in pauli_sum.coeffs)
```

This should be quite easy to implement.

https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/opflow/primitive_ops/pauli_sum_op.py

### PR DESCRIPTION
**Pull request by paolob67**
### Summary
Fixes #5547
- Added the coeffs property to return the operator coefficients  as a numpy array
- Added a matrix iterator by leveraging the same code used in the PauliSparseOperator
- Added tests for the iterator

### Details and comments
This fix provides the ability of being able to iterate over, or return a list of coefficients in PauliSumOp. At present, one has to dig into the details to construct the coefficients for each term. In particular, it requires multiplying two numbers for each coeffcient. There are use cases, for instance when bounding the eigenvalues of the operators.
The coeffs property returns a numpy array of the operator obtained by multiplying the coeffs of the primitive by the coefficient provided in the constructor.
The matrix iterator iterates through the Pauli table representing the the PauliSumOp in either sparse or not fashion.

### CODE REVIEW

**Review by ikkoham — COMMENTED** [review]
Very nice. Just a small comment.

File: test/python/opflow/test_pauli_sum_op.py [hunk 1]
```diff
@@ -33,7 +34,24 @@
     Zero,
 )
 from qiskit.circuit import Parameter, ParameterVector
-from qiskit.quantum_info import Pauli, SparsePauliOp
+from qiskit.quantum_info import Pauli, SparsePauliOp, PauliTable
+
+
+def pauli_mat(label):
```
[the remark this answers is not in the document]
Comment (paolob67) [reply test/python/opflow/test_pauli_sum_op.py:40]: Yes, makes sense. I'll commit a new version with the suggested change. Going through tests right now on my env.
Thanks a lot.

**Comment by paolob67** [comment]
The `Qiskit.qiskit-terra (Python_Tests Tutorials Python38) ` pipeline is failing after the commit on the PR but I really don't know how to fix that. Any help is appreciated. Thanks.

**Comment by jlapeyre** [comment]
> pipeline is failing after the commit on the PR

It may be unrelated to this PR. There is probably a way to re-run it.

**Comment by kdk** [comment]
> > pipeline is failing after the commit on the PR
> 
> It may be unrelated to this PR. There is probably a way to re-run it.

I triggered a re-run and it passed. Maybe a transient azure issue?

**Review by woodsp-ibm — APPROVED** [review]

**Pull request merged**